# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Wed Jan 13 20:14:54 2021 +0000
+++ b/dom/canvas/WebGLContext.h	Wed Jan 13 21:49:40 2021 +0000
@@ -1237,35 +1237,7 @@
 
   template <typename... Args>
   void GeneratePerfWarning(const char* const fmt, const Args&... args) const {
-    if (!ShouldGeneratePerfWarnings()) return;
-
-    const auto funcName = FuncName();
-    nsCString msg;
-    msg.AppendPrintf("WebGL perf warning: %s: ", funcName);
-
-#ifdef __clang__
-#  pragma clang diagnostic push
-#  pragma clang diagnostic ignored "-Wformat-security"
-#elif defined(__GNUC__)
-#  pragma GCC diagnostic push
-#  pragma GCC diagnostic ignored "-Wformat-security"
-#endif
-    msg.AppendPrintf(fmt, args...);
-#ifdef __clang__
-#  pragma clang diagnostic pop
-#elif defined(__GNUC__)
-#  pragma GCC diagnostic pop
-#endif
-
-    GenerateErrorImpl(0, msg);
-
-    mNumPerfWarnings++;
-    if (!ShouldGeneratePerfWarnings()) {
-      GenerateWarning(
-          "After reporting %u, no further WebGL perf warnings will"
-          " be reported for this WebGL context.",
-          uint32_t(mNumPerfWarnings));
-    }
+    GenerateError(webgl::kErrorPerfWarning, fmt, args...);
   }
 
  public: